### PR TITLE
Version 0.3.0

### PR DESCRIPTION
**Pull request by Henley13**
TODO:
- [x] add a readme
- [x] initiate test coverage
- [x] complete documentation bigfish.stack
- [x] complete documentation bigfish.plot
- [x] complete documentation bigfish.detection
- [x] complete documentation bigfish.segmentation
- [x] add notebook examples preprocessing
- [x] add notebook examples segmentation
- [x] add notebook examples detection
- [x] add notebook examples postprocessing
- [x] refactor bigfish.stack
- [x] refactor bigfish.plot
- [x] refactor bigfish.segmentation
- [x] refactor bigfish.detection
- [x] add 2-d cytoplasm segmentation
- [x] add focus projection
- [x] add foci decomposition

OPTIONAL:
- [ ] ensure test coverage
- [ ] add CIDRE illumination correction
- [ ] add network robustness quantification
- [ ] add Inception model
- [ ] add ResNet model
- [ ] add Unet model
- [ ] add DL classification pattern (pretrained model)
- [ ] add RF classification pattern (pretrained model)
- [ ] complete documentation bigfish.classification
- [ ] add notebook examples analysis
- [ ] refactor bigfish.classification

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+--index-url https://pypi.python.org/simple/
+
+-e .
```
Comment (oeway) [inline requirements.txt:3]: I would perhaps remove this from the requirements, since the requirements is typically about requiring other packages.

**Comment by oeway** [comment]
A suggestion here, I think nowadays it's more of a convention to use semantic versioning: https://semver.org/ , so instead of going straight to Version 1.0, you can start from 0.1.0, then increase the numbers according to semantic versioning. According to that, I would keep the version 1.0.0 for later when we want to fix the public api.

**Comment by Henley13** [comment]
> A suggestion here, I think nowadays it's more of a convention to use semantic versioning: https://semver.org/ , so instead of going straight to Version 1.0, you can start from 0.1.0, then increase the numbers according to semantic versioning. According to that, I would keep the version 1.0.0 for later when we want to fix the public api.

Maybe 0.3.0 ? I would label my two tags paper_racha and qbi_2020 as 0.1.0 and 0.2.0 respectively... If I find a proper way to do that. Otherwise, I agree with 0.1.0.

File: requirements.txt (hunk 1 above)
Comment (Henley13) [reply requirements.txt:3]: Yes, casual user does not need editable mode (-e .) and a dev can add it by himself.

**Comment by muellerflorian** [comment]
I think either way (0.1.0 or 0.3.0) is fine. My slight preference would also be 0.3.0 to better illustrate how the repo evolved.

**Comment by muellerflorian** [comment]
another comment. I think we should **provide test data** somewhere to be downloaded. like this, the test examples could be tried with the same data and results be compared.

**Pull request merged**